### PR TITLE
DT-945: Handle Update Study Registration Better

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-945

### Summary
This PR makes some changes to the Update Study API to fix a few bugs. As a result of updating a production study (see [DT-941](https://broadworkbench.atlassian.net/browse/DT-941)), there was a non-trivial amount of hand-editing the JSON to get around incorrect validation rules. This PR fixes that to make updates a lot more smoother in the future.

* Phenotype is not a required field
* Data Custodian Email  is not a required field
* If the consent group (i.e. dataset) is being updated:
  * All data use related fields are ignored
  * Access management changes are ignored
* If the consent group (i.e. dataset) is a new entity, all data use and access management values are preserved.
* Everything else provided in the study is valid for update.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

[DT-941]: https://broadworkbench.atlassian.net/browse/DT-941?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/DatasetRegistrationSchemaV1UpdateValidator.java [hunk 1]
```diff
@@ -20,6 +27,89 @@ public DatasetRegistrationSchemaV1UpdateValidator(DatasetService datasetService)
...
+    public boolean shouldSkipClass(Class<?> aClass) {
+      return aClass.getSimpleName().equalsIgnoreCase("ConsentGroup");
+    }
+  };
+
+  private final ExclusionStrategy consentGroupExclusionStrategy = new ExclusionStrategy() {
+    @Override
+    public boolean shouldSkipField(FieldAttributes fieldAttributes) {
+      final HashSet<String> exclusions = new HashSet<>(List.of(
+          "accessManagement",
```
Comment (rjohanek) [inline src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/DatasetRegistrationSchemaV1UpdateValidator.java]: are data use fields an enum somewhere?

**Review by rjohanek — APPROVED** [review]
this seems like a great improvement to make updating studies easier for users in the future!

File: src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/DatasetRegistrationSchemaV1UpdateValidator.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/DatasetRegistrationSchemaV1UpdateValidator.java]: Aha ... they are defined as static strings in `DatasetRegistrationSchemaV1Builder`, thank you! Will make that update.

**Review by snf2ye — APPROVED** [review]
I'm having a hard time wrapping my head around using GSON to convert between json and java objects, but as far as I can tell, LGTM!

**Comment by rushtong** [comment]
> I'm having a hard time wrapping my head around using GSON to convert between json and java objects, but as far as I can tell, LGTM!

To be honest, this was super painful. If I were to re-implement this, I might choose a very different set of updates to support.

**Pull request merged**